### ISSUE TITLE
First instance of recurring transaction not firing automatically or manually

### ISSUE THREAD

**Issue by makobot-sh** [firefly-iii/firefly-iii#9466]
### Support guidelines

- [x] I've read the [support guidelines](https://github.com/firefly-iii/firefly-iii/blob/main/.github/support.md)

### I've found a bug and checked that ...

- [x] ... [the documentation](https://docs.firefly-iii.org/) does not mention anything about my problem
- [x] ... there are no open or closed issues that are related to my problem
- [x] ... it's [definitely a Firefly III issue, not me](https://github.com/firefly-iii/firefly-iii/blob/main/.github/its_you_not_me.md)

### Description

I am testing out firefly iii locally on ubuntu using the provided docker-compose. So far, everything has been working as expected. I created some monthly recurring transactions and was hoping to see their first iteration impact a couple days later on nov. 12, but the transactions were not created even though I have the cron job container running and the logs seemed to indicate the job was fired successfully:
```sh
2024-11-12 11:00:00 crond: USER root pid  40 cmd wget -qO- http://app:8080/api/v1/cron/8fe5a982783959d81f95e343e3910661;echo
2024-11-12 11:00:00 {"recurring_transactions":{"job_fired":true,"job_succeeded":true,"job_errored":false,"message":"Recurring transactions cron job fired successfully."},"auto_budgets":{"job_fired":true,"job_succeeded":true,"job_errored":false,"message":"Auto-budget cron job fired successfully."},"bill_warnings":{"job_fired":true,"job_succeeded":true,"job_errored":false,"message":"Bill warning cron job fired successfully."}}
```

I tried creating a new recurring transaction for nov. 14th and firing the first "expected withdrawal" manually, but i got the error **"Message: No new transaction was created. Perhaps it was already fired for this date?"**. I tried to fire the second expected withdrawal afterwards and that one was created successfully. 

I was able to reproduce this behaviour in the demo page, where I took the following screenshots:

![Error message when trying to fire first expected withdrawal of sample recurring transction](https://github.com/user-attachments/assets/8b6b0646-2457-4373-afa6-85de117b0bbe)
![Success when firing second expected withdrawal of sample recurring transaction](https://github.com/user-attachments/assets/2a62e819-3fce-44c3-a59d-ddcac26c4b19)
![Sample recurring transaction](https://github.com/user-attachments/assets/0cd4f76c-e10a-4a1c-9a24-7a19f0249221)

I can't confirm whether the cron jobs would work on the second iteration of my recurring transactions, I haven't been able to figure out how to run the cron jobs on a spoofed date on the docker setup (I've read of --force date but I couldn't figure out how to translate that to the cron docker setup). Currently crontab on my firefly_iii_cron container looks like this:
```
0 14 * * * wget -qO- http://app:8080/api/v1/cron/8fe5a982783959d81f95e343e3910661;echo
```
(14 corresponds to 11hs on the previous cron logs since there's a mismatch between my timezone (GMT-3) and the container timezone (UTC))

If there is a way to force date through wget please let me know!

### Debug information

Debug information generated at 2024-11-13 17:24:26 for Firefly III version **v6.1.22**.

<table><thead><tr><th colspan="2">System information</th></tr><tr><th>Item</th><th>Value</th></tr></thead><tbody><tr><td>Firefly III</td><td>6.1.22 / v2.1.0 / 25 (exp. 25)</td></tr><tr><td>PHP version</td><td>8.3.13 (64bits) / apache2handler / Linux x86_64</td></tr><tr><td>BCscale</td><td>12</td></tr><tr><td>Error reporting</td><td>Display: Off, reporting: ALL errors</td></tr><tr><td>Max upload</td><td>67108864 (64 MB)</td></tr><tr><td>Database drivers</td><td><strong>*mysql*</strong>, pgsql, sqlite, </td></tr><tr><td>Docker build</td><td><span>#</span>1147, base <span>#</span>92</td></tr></tbody></table><table><thead><tr><th colspan="2">Firefly III information</th></tr><tr><th>Item</th><th>Value</th></tr></thead><tbody><tr><td>Timezone</td><td>Europe/Amsterdam + America/Argentina/Buenos_Aires</td></tr><tr><td>App environment</td><td>production, debug: false</td></tr><tr><td>Layout</td><td>v1</td></tr><tr><td>Logging</td><td>notice, stack / (empty)</td></tr><tr><td>Cache driver</td><td>file</td></tr><tr><td>Default language and locale</td><td>en_US + equal</td></tr><tr><td>Trusted proxies</td><td></td></tr><tr><td>Login provider & user guard</td><td>eloquent / web</td></tr><tr><td>Login headers</td><td>N/A + N/A</td></tr><tr><td>Stateful domains</td><td></td></tr><tr><td>Last cron job</td><td>2024-11-13 23:00:00 (6 hours from now)</td></tr><tr><td>Mailer</td><td>log</td></tr></tbody></table><table><thead><tr><th colspan="2">User-specific information</th></tr><tr><th>Item</th><th>Value</th></tr></thead><tbody><tr><td>User</td><td><span>#</span>1 of 1</td></tr><tr><td>User flags</td><td><span title="Has liabilities">:credit_card:</span> <span title="Has recurring transactions">:clock130:</span></td></tr><tr><td>Session start</td><td>2024-11-01 00:00:00</td></tr><tr><td>Session end</td><td>2024-11-30 23:59:59</td></tr><tr><td>View range</td><td>1M</td></tr><tr><td>User language</td><td>en_US</td></tr><tr><td>User locale</td><td>en_US</td></tr><tr><td>Locale(s) supported</td><td>en_US.utf8: :white_check_mark:<br>en_US.UTF-8: :white_check_mark:<br></td></tr><tr><td>User agent</td><td>Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:132.0) Gecko/20100101 Firefox/132.0</td></tr></tbody></table>

### Expected behaviour

When firing the first expected transaction manually, the corresponding transaction should be created with current date. The button to "Create right now" for that transaction should disappear, and a cron job ran in the corresponding date should not re-create the transaction.

### Steps to reproduce

1. Create a new recurring transaction. Have it repeat monthly and have a fixed number of repetitions.
2. Enter the newly created transaction and try to fire the first expected withdrawal manually.
3. **"Message: No new transaction was created. Perhaps it was already fired for this date?"** should appear and no new transaction be created
4. Fire the second expected withdrawal manually
5. The transaction should be created properly

**Comment by JC5** [comment]
Nice find, and thanks for an excellent bug report!

This will be fixed in the next release. The error was in one letter 😅 .

**Comment by makobot-sh** [comment]
Awesome, thanks for the quick response! If it's not too much of a bother, can I ask again if there's a way to spoof the date of the cron jobs with the docker setup? I've been looking into changing the date of the container directly but maybe there's an easier route

**Comment by JC5** [comment]
Spoofing the date one time is the easiest way to do this. 

Something like "docker exec -it fireflyiiicontainer bash"

Then you get a prompt and you can do "php artisan firefly:cron --date=2024-11-11"

**Comment by makobot-sh** [comment]
That worked, thank you!!

**Comment by shanelynn321** [comment]
It seems this may still be an issue. Can anyone else confirm they have this working as intended? 

I have confirmed that the cron job is indeed firing as it is supposed to. I have used the cron container built into the docker compose as well as taken that container out and run it directly using the external url in the base os cron. 

I can manually create the transactions, however when the cron job fires, it does not create the transactions.
